.github/workflows/lockfiles.yaml

```diff
@@ -0,0 +1,594 @@
+          # Font install based on: https://github.com/JetBrains/JetBrainsMono/blob/v2.304/install_manual.sh
+          gh release download "v2.304" \
+            -R "JetBrains/JetBrainsMono" \
+            --pattern "JetBrainsMono-*.zip" \
+            --output "${RUNNER_TEMP}/JetBrainsMono.zip"
+          mkdir -p ~/.local/share/fonts
+          unzip -o "${RUNNER_TEMP}/JetBrainsMono.zip" -d ~/.local/share/fonts
+          fc-cache -f
+
+      - name: Install freeze
+        if: steps.lockfile.outputs.CHANGED == 'true'
+        env:
+          VERSION: '0.2.2'
+          GOARCH: ${{ fromJSON('{"X86":"i386","X64":"x86_64","ARM":"arm","ARM64":"arm64"}')[runner.arch] }}
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release download "v${VERSION}" \
+            -R "charmbracelet/freeze" \
+            --pattern "freeze_*_${RUNNER_OS}_${GOARCH}.tar.gz" \
+            --output "${RUNNER_TEMP}/freeze.tar.gz"
+          tar xzf "${RUNNER_TEMP}/freeze.tar.gz" \
+            -C ~/.local/bin \
+            --strip-components=1 \
+            --wildcards \
+            '*/freeze'
+          [[ -x ~/.local/bin/freeze ]] || chmod +x ~/.local/bin/freeze
+
+      - name: Freeze lockfile diff as picture
+        if: steps.lockfile.outputs.CHANGED == 'true'
+        # For samples of the themes freeze can use when generating a "terminal screenshot", see:
+        #   - "charm" theme (default): https://github.com/charmbracelet/freeze
+        #   - all other themes: https://xyproto.github.io/splash/docs/
+        env:
+          # Freeze processes the output line-by-line, so the ansi escape sequences
+          # that span multiple lines only apply to the first line.
+          # This sed script repeats those ansi escape sequences on each line.
+          SED_SCRIPT: |
+            s/^\x1B\[4m \+$/\0\x1B[0m/     # append ansi reset on line above heading
+            s/^== .* ==$/\x1B[4m\0\x1B[0m/ # add ansi underline to headling
+            s/^\x1b\[0m$//                 # drop ansi reset after the heading
+        run: |
+          for theme in github github-dark; do
+            for ext in svg png; do
+              # The diff output applies ansi underlines across multiple lines, but freeze formats
+              # each line separately. So, use sed to repeat ansi chars per line.
+              sed -e "${SED_SCRIPT}" "${STDERR_LOG}" \
+                | freeze --config full --language ansi --theme "${theme}" --width 720 \
+                --output "${STDERR_LOG}.${theme}.${ext}" -
+            done
+          done
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v5.0.0
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-session-name: GitHubActions-${{ github.run_id }}
+          role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/GitHubActions@org=${{ github.repository_owner }},repo=${{ github.event.repository.name }}
```

Comment:
I ended up with this `@org=...,repo=...` because role names cannot contain `/` characters.